contracts/VaultV3.vy

```diff
@@ -650,10 +639,15 @@ def _process_report(strategy: address) -> (uint256, uint256):
         self.strategies[strategy].total_gain += gain
         # update current debt after processing management fee
         self.strategies[strategy].current_debt += gain
-        self.locked_profit = self._calculate_locked_profit() + gain - total_fees
 
+        gain_without_fees: uint256 = gain - total_fees
+        new_profit_end_date: uint256 = block.timestamp + (pending_profit * remaining_time + gain_without_fees * self.profit_max_unlock_time) / (pending_profit + gain_without_fees)
+        self.profit_distribution_rate = (pending_profit + gain_without_fees) * MAX_BPS / (new_profit_end_date - block.timestamp)
```

Comment:
warning: if `new_profit_end_date == block.timestamp` (which could happen if the Accounting Manager decides to take profit immediately (perfectly valid), this would revert (division by 0). 

so this needs a check and handling the edge case (profit is immediately unlocked by adding it `total_debt`)

Reply:
adding code for that specific edge case

Reply:
I will actually handle the edge case as if `profit_max_unlock_time==0` which means that we don't want any profit smoothing